unlock-app/src/reducers/lockFormVisibilityReducer.ts

```diff
@@ -0,0 +1,30 @@
+import { SHOW_FORM,
+         HIDE_FORM,
+} from '../actions/lockFormVisibility'
+import { SET_PROVIDER } from '../actions/provider'
+import { SET_NETWORK } from '../actions/network'
+
+export const initialState = {
+  visible: false,
+}
+
+const lockFormVisibilityReducer = (state = initialState,
+                                   action: { type: string }) => {
+  if ([SET_PROVIDER, SET_NETWORK].indexOf(action.type) > -1) {
```

Comment:
I think we also need `SET_ACCOUNT` here?

Reply:
Oh, I think you're right. I used the accounts reducer as my template which handled that separately, but if I'd picked any other one I'd have gotten it right. I'll add that one in.